client/adventureMap/CList.cpp

```diff
@@ -301,6 +305,72 @@ void CHeroList::CHeroItem::gesture(bool on, const Point & initialPosition, const
 	GH.windows().createAndPushWindow<RadialMenu>(pos.center(), menuElements, true);
 }
 
+bool CHeroList::CHeroItem::captureThisKey(EShortcut key)
+{
+
+	switch(key)
+	{
+	case EShortcut::LIST_HERO_UP:
+	case EShortcut::LIST_HERO_DOWN:
+	case EShortcut::LIST_HERO_TOP:
+	case EShortcut::LIST_HERO_BOTTOM:
+	case EShortcut::LIST_HERO_DELETE:
+		return hero && pos.isInside(GH.getCursorPosition());
+	}
+	return false;
+}
+
+void CHeroList::CHeroItem::keyPressed(EShortcut key)
+{
+	if(!hero)
+		return;
+
+	if(!pos.isInside(GH.getCursorPosition()))
+		return;
+
+	auto & heroes = LOCPLINT->localState->getWanderingHeroes();
+
+	if(key == EShortcut::LIST_HERO_DELETE)
+	{
+		CFunctionList<void()> ony = [=](){ };
+		ony += [=](){ LOCPLINT->cb->dismissHero(hero); };
+		LOCPLINT->showYesNoDialog(CGI->generaltexth->allTexts[22], ony, nullptr);
+		return;
+	}
+
+	if(heroes.size() < 2)
+		return;
+
+	size_t heroPos = vstd::find_pos(heroes, hero);
+	const CGHeroInstance * heroUpper = (heroPos < 1) ? nullptr : heroes.at(heroPos - 1);
```

Comment:
Can we use boolean isHeroUpper here instead? Or can heroes.at(index) contain nullptr itself?

Reply:
Not sure. Used the same line as for the gesture here. To be safe.